### DISCUSSION TITLE
How To: Test Changes On The Website

### DISCUSSION THREAD

**Discussion by TeaBizzy** [Energy-Price-News-API/energy-prices-api#137] [Q&A] [answered]
Hi everyone, I'm just learning the ropes of web dev, I'm not sure what I'm doing wrong but I can't seem to test the website on my local machine.

I've followed the `README.md` instructions.
`npm install` works just fine but when I try `npm run dev` I get the following, and it just hangs:
![Screenshot from 2022-09-22 10-03-16](https://user-images.githubusercontent.com/111951212/191769203-f0aa6a15-b15b-4839-b964-7eb15c1b185f.png)

I assume thats supposed to build/host the website locally, and then I can just type localhost:8000 into my browser and access the page, but I get the following:

![Screenshot from 2022-09-22 10-03-43](https://user-images.githubusercontent.com/111951212/191769534-753ba440-206f-4ccd-bcb9-3c2f6a31615f.png)

Seeing that isn't working, I found `index.html` and opened it up, but any links that don't go to github also don't work. Comparing this to the actual site, there's a massive difference.

![Screenshot from 2022-09-22 10-10-12](https://user-images.githubusercontent.com/111951212/191770346-94e01941-a662-4737-8516-8350b086cbd0.png)

If anyone can tell me where I'm going wrong and assist a newbie I would be willing to listen and learn! Hopefully I could contribute more in the future after I'm up to speed! :smiley: 

Thank you!

**Comment by takanome-dev** [comment]
Hey @TeaBizzy I don't know why it says that `keyword.replaceAll` is not a function, I'm the one who implemented that :sweat_smile: but I'm not getting that error :thinking: .

We still have some issues with Twitter search and it crashes the server, but try to restart the server and test again. If it still crashes you can open an issue for that, get assigned to work on it and contribute to the project :rocket:

**Comment by TeaBizzy** [comment]
Hi @TAKANOME-DEV. I tried to run `npm run dev` again and still get the same error. I'll draft up an issue for it. I think I have an idea how to fix it. Just strange that this is happening with newly cloned repos?

**Comment by MizouziE** [comment, marked as answer]
Hi @TeaBizzy , can  you let me know what version of node you have installed?
```shell
node -v
```
Should give you the version.

For example, this project was started on v16
![image](https://user-images.githubusercontent.com/90829439/191867292-1240d481-15e8-4c40-87c1-a1433df9877c.png)

I have cloned a fresh copy of the repo and set it up with the same steps as are written and it worked just fine right off the bat.

On doing some research I came across [this article explaining the issue](https://www.designcise.com/web/tutorial/how-to-fix-replaceall-is-not-a-function-javascript-error)  and also [this article](https://backbencher.dev/articles/javascript-es2021-new-features) which made it clear that the function is fairly new. This makes me think that perhaps the version of node you're running is older than this.

The same thing could be causing the other issues you are seeing, so for now I won't merge. What I believe is happening is:
- Nodemon doesn't start up properly as it is supposed to point to `./src/app.js` which in turn points to the routes (accessing the homepage, you're asking for the `/` route), which then points to the controller for the home route, which then points to the `./src/views/index.html`
- Technically you're accessing this from the root, just redirected through a bunch of other files, so the current reference to `/ccc/styles.css` works as expected because the browser will only look in the `./public` folder.
- By manually accessing the `index.html` the link is treated as relative **to** the current file and that doesn't exist. It all working, depends on the server observing the structure and forwarding we have in place. Circumventing that will cause what you're seeing.

Check your node version, I'm pretty certain it will be an older version. If it is, upgrade it to the latest stable version or install it as well as your current version if you have project which rely on it, then run this project on the newer node as it currently is in our main repo.

I definitely need to specify a Node.js version in the README.md. I appreciate you bringing this to my attention.

I genuinely thought that the `package.json` would make any system running lower than 16 squawk...
![image](https://user-images.githubusercontent.com/90829439/191869531-d57793e6-73fa-4fd7-8ff6-00c14e51878b.png)

But maybe it doesn't.

**Reply by TeaBizzy** [reply]
Hi @MizouziE, thank you for the detailed reply. Turns out I had node 12.22.12 installed and wasn't aware this was the issue. Sorry for any hassle I caused trying to fix this with the PR's.

Cheers :clinking_glasses:

**Reply by MizouziE** [reply]
No problem at all bro! We're all here to learn about web development in the real world and I learnt something new in investigating this issue. 

I really appreciate that you asked and I hope folks in the future see your example and do the same. 👊
